### DISCUSSION TITLE
createMany() & relations : is connect possible?

### DISCUSSION THREAD

**Discussion by rwmd** [prisma/prisma#23786] [Q&A] [answered]
### Question

As stated in the the [documentation](https://www.prisma.io/docs/orm/prisma-client/queries/relation-queries#create-multiple-records-and-multiple-related-records), it is not possible to CREATE relations with createMany().

The documentation does not state the behaviour for connecting to existing records.

The same payload works for create() but gives error-messages for createMany():

       [...,
         {
           ts: new Date("2024-04-09T09:03:07.721Z"),
           topic: {
             connect: {
               id: 1
             }
           },
           job: {
             connect: {
               id: 1
             }
           },
           status: "W",
           content: "Warning 16",
           js: "{\"empty\":\"\"}"
         }
       ]

"Unknown argument `topic`. Did you mean `topicId`? Available options are marked with ?."

It's against the logic for me, since there is no nested create/write neccessary for this.
Do I do something wrong or is this intended behaviour?

best regards,
R. 

### Expected behavior (optional)

createMany() with "connect" for relations

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model Topic {
id          Int         @id @default(autoincrement())
name        String      @db.VarChar(40) @unique
jobs        Job[]
logs        Log[]
lastJobId   Int?
lastStart   DateTime?
locked      Boolean     @default(false)
}

model Job {
  id          Int         @id @default(autoincrement())
  name        String      @db.VarChar(40) @unique
  status      String      @db.Char(1)
  tsStart     DateTime
  tsEnd       DateTime?
  runTime     Int?
  info        String?     @db.VarChar(200)
  js          Json?       @db.Json
  topic       Topic?      @relation(fields: [topicId], references: [id])
  topicId     Int?
  logs        Log[]
}

model Log {
  id          Int         @id @default(autoincrement())
  ts          DateTime
  topic       Topic?      @relation(fields: [topicId], references: [id])
  topicId     Int?
  job         Job?        @relation(fields: [jobId], references: [id])
  jobId       Int?
  status      String?     @db.VarChar(1)
  content     String?     @db.VarChar(200)
  js          Json?       @db.Json
}
```

```
    const aLog = [{ ts: new Date(), topic: { connect: { id: oTopic.id } }, job: { connect: { id: oJob.id } }, status: 'W', content: 'Warning 9', js: JSON.stringify({ empty: '' }) },
    { ts: new Date(), topic: { connect: { id: oTopic.id } }, job: { connect: { id: oJob.id } }, status: 'W', content: 'Warning 11', js: JSON.stringify({ empty: '' }) },
    { ts: new Date(), topic: { connect: { id: oTopic.id } }, job: { connect: { id: oJob.id } }, status: 'W', content: 'Warning 12', js: JSON.stringify({ empty: '' }) },
    { ts: new Date(), topic: { connect: { id: oTopic.id } }, job: { connect: { id: oJob.id } }, status: 'W', content: 'Warning 13', js: JSON.stringify({ empty: '' }) },
    { ts: new Date(), topic: { connect: { id: oTopic.id } }, job: { connect: { id: oJob.id } }, status: 'W', content: 'Warning 14', js: JSON.stringify({ empty: '' }) },
    { ts: new Date(), topic: { connect: { id: oTopic.id } }, job: { connect: { id: oJob.id } }, status: 'W', content: 'Warning 15', js: JSON.stringify({ empty: '' }) },
    { ts: new Date(), topic: { connect: { id: oTopic.id } }, job: { connect: { id: oJob.id } }, status: 'W', content: 'Warning 16', js: JSON.stringify({ empty: '' }) }]

    dbg('addLog', await prisma.log.create({data: aLog[0]}))
    dbg('addLogMany', await prisma.log.createMany({data: aLog}))



```

- OS: Windows, Linux
- Database: <PostgreSQL
- Node.js version: v20.11.1
- Prisma version: 5.12.1

**Comment by ludralph** [comment, marked as answer]
Hi @rwmd 👋 

Thank you for raising this question.

Can you modify the payload to pass in the foreign key id directly instead of using `connect`?
```js
 [...,
     {
       ts: new Date("2024-04-09T09:03:07.721Z"),
       topicId,
       jobId,
       status: "W",
       content: "Warning 16",
       js: "{\"empty\":\"\"}"
     }
   ]
```
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you f

**Reply by rwmd** [reply]
Thanks, this does the trick.
Even easier, maybe worth putting it into the docs?
